File: examples/javascript/poll-video.mjs
import { USVideoAPI } from "../../sdk/javascript/index.mjs";

const client = new USVideoAPI({ apiKey: process.env.USVIDEOAPI_KEY });

const job = await client.createVideo({
  prompt:
    "A 9:16 salon ad with polished service shots, quick transformations, and a booking CTA.",
});

console.log("created", job);

const result = await client.waitForVideo(job.id, {
  pollIntervalMs: 5000,
  timeoutMs: 300000,
});

console.log("result", result);

File: examples/python/poll_video.py
import os

from sdk.python.usvideoapi import USVideoAPI


client = USVideoAPI(api_key=os.environ["USVIDEOAPI_KEY"])

job = client.create_video(
    prompt="A 9:16 medspa ad with before/after framing, benefit callouts, and a consultation CTA."
)
print("created", job)

result = client.wait_for_video(job["id"], poll_interval=5, timeout=300)
print("result", result)
